2d project without `default-features` fails to compile without `bevy_pbr` feature on wasm targets

## Bevy version

main, bisected to #10057

## Relevant system information

```
AdapterInfo { name: "Apple M1 Max", vendor: 0, device: 0, device_type: IntegratedGpu, driver: "", driver_info: "", backend: Metal }
SystemInfo { os: "MacOS 14.2.1 ", kernel: "23.2.0", cpu: "", core_count: "10", memory: "64.0 GiB" }
```

## What you did

```bash
cargo run --example sprite --no-default-features \
--features=bevy_asset,bevy_core_pipeline,bevy_render,bevy_sprite,bevy_text,bevy_ui,bevy_winit,multi-threaded,webgl2,x11,png \
--target=wasm32-unknown-unknown
```

(adding `bevy_pbr` to that list causes compilation to succeed)

## What went wrong

```
error[E0599]: no variant or associated item named `Glsl` found for enum `ShaderLanguage` in the current scope
   --> crates/bevy_render/src/render_resource/shader.rs:217:70
    |
217 |             Source::Glsl(_, _) => naga_oil::compose::ShaderLanguage::Glsl,
    |                                                                      ^^^^ variant or associated item not found in `ShaderLanguage`

error[E0599]: no variant or associated item named `GlslVertex` found for enum `naga_oil::compose::ShaderType` in the current scope
   --> crates/bevy_render/src/render_resource/shader.rs:233:77
    |
233 |                 naga::ShaderStage::Vertex => naga_oil::compose::ShaderType::GlslVertex,
    |                                                                             ^^^^^^^^^^ variant or associated item not found in `ShaderType`

error[E0599]: no variant or associated item named `GlslFragment` found for enum `naga_oil::compose::ShaderType` in the current scope
   --> crates/bevy_render/src/render_resource/shader.rs:234:79
    |
234 |                 naga::ShaderStage::Fragment => naga_oil::compose::ShaderType::GlslFragment,
    |                                                                               ^^^^^^^^^^^^ variant or associated item not found in `ShaderType`

For more information about this error, try `rustc --explain E0599`.
error: could not compile `bevy_render` (lib) due to 3 previous errors
```

## Workaround

Users can work around this by adding the following to their `Cargo.toml`

```toml
[target.'cfg(target_arch = "wasm32")'.dependencies]
# Workaround for Bevy #11908
naga_oil = "*"
```

--- a/crates/bevy_internal/Cargo.toml
+++ b/crates/bevy_internal/Cargo.toml
@@ -58,10 +58,7 @@ symphonia-vorbis = ["bevy_audio/symphonia-vorbis"]
 symphonia-wav = ["bevy_audio/symphonia-wav"]
 
 # Shader formats
-shader_format_glsl = [
-  "bevy_render/shader_format_glsl",
-  "bevy_pbr?/shader_format_glsl",
-]
+shader_format_glsl = ["bevy_render/shader_format_glsl"]
 shader_format_spirv = ["bevy_render/shader_format_spirv"]
 
 serialize = [

--- a/crates/bevy_pbr/Cargo.toml
+++ b/crates/bevy_pbr/Cargo.toml
@@ -11,7 +11,6 @@ keywords = ["bevy"]
 [features]
 webgl = []
 webgpu = []
-shader_format_glsl = ["naga_oil/glsl"]
 pbr_transmission_textures = []
 
 [dependencies]
@@ -38,15 +37,6 @@ bytemuck = { version = "1", features = ["derive"] }
 radsort = "0.1"
 smallvec = "1.6"
 thread_local = "1.0"
-
-[target.'cfg(target_arch = "wasm32")'.dependencies]
-naga_oil = "0.13"
-
-[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
-# Omit the `glsl` feature in non-WebAssembly by default.
-naga_oil = { version = "0.13", default-features = false, features = [
-  "test_shader",
-] }
 
 [lints]
 workspace = true

--- a/crates/bevy_render/Cargo.toml
+++ b/crates/bevy_render/Cargo.toml
@@ -73,9 +73,6 @@ wgpu = { version = "0.19.1", default-features = false, features = [
   "fragile-send-sync-non-atomic-wasm",
 ] }
 naga = { version = "0.19", features = ["wgsl-in"] }
-naga_oil = { version = "0.13", default-features = false, features = [
-  "test_shader",
-] }
 serde = { version = "1", features = ["derive"] }
 bitflags = "2.3"
 bytemuck = { version = "1.5", features = ["derive"] }
@@ -98,7 +95,15 @@ profiling = { version = "1", features = [
 ], optional = true }
 async-channel = "2.1.0"
 
+
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+# Omit the `glsl` feature in non-WebAssembly by default.
+naga_oil = { version = "0.13", default-features = false, features = [
+  "test_shader",
+] }
+
 [target.'cfg(target_arch = "wasm32")'.dependencies]
+naga_oil = "0.13"
 js-sys = "0.3"
 # web-sys doesn't follow semver for the WebGPU APIs as they are unstable
 # Make sure that WebGPU builds work when changing this!

